torch/fx/experimental/sym_node.py

```diff
@@ -1645,6 +1652,86 @@ def unary_magic_impl(self):
             return (method_to_operator(method))(get_constant(self))
         return wrap_node(getattr(self.node, method_attr)())
 
+    def uninteresting_files() -> Set[str]:
+        import inspect
+
+        import torch
+
+        mods = [
+            torch._dynamo.eval_frame,
+            torch._dynamo.utils,
+            torch.fx.experimental.sym_node,
+            torch,
+        ]
+        import torch._dynamo.guards
+
+        return (
+            {inspect.getfile(m) for m in mods}
+            | torch._dynamo.guards.uninteresting_files()
+            | {"<string>"}
+        )
+
+    def capture_provenance(fn):
+        @functools.wraps(fn)
+        def wrapper(self, other):
+            result = fn(self, other)
+            if torch._logging._internal.GET_DTRACE_STRUCTURED:
+                floc = None
+                user_stack = None
+                user_top_stack = None
+                user_bottom_stack = None
+
+                if len(TracingContext.extract_stack()) > 0:
+                    user_stack = TracingContext.extract_stack()
+                    user_top_stack = format_frame(user_stack[0], line=True)
+                    user_bottom_stack = format_frame(user_stack[-1], line=True)
+
+                frame = inspect.currentframe()
+                try:
+                    while frame is not None:
+                        if (
+                            floc is None
+                            and frame.f_code.co_filename not in uninteresting_files()
+                        ):
+                            floc = format_frame(
+                                traceback.FrameSummary(
+                                    frame.f_code.co_filename,
+                                    frame.f_lineno,
+                                    frame.f_code.co_name,
+                                ),
+                                line=True,
+                            )
+                        if frame.f_back is None and user_top_stack is None:
+                            user_top_stack = format_frame(
+                                traceback.FrameSummary(
+                                    frame.f_code.co_filename,
+                                    frame.f_lineno,
+                                    frame.f_code.co_name,
+                                ),
```

Comment:
Would it be possible to reuse trace_structured's `from_traceback` function?

Reply:
hmm from_traceback gives you a `list[dict[str, Any]]` but I want a formatted string. are you suggesting serializing the dict?

Reply:
Yeah, for example like [create_unbacked_symbol](https://github.com/pytorch/pytorch/blob/15e12d5ec392a598a4c8e77d5bd95c3c1727a462/torch/fx/experimental/symbolic_shapes.py#L4232-L4242), and most other places that log to trace_structured. 

Also separately if we could match how the two stacks are formatted, it'll make it easier to combine these two information together -- the provenance trie could then go down to the logs for where the unbacked symbols are created.